pkg/providers/v1/aws_utils.go

```diff
@@ -43,3 +47,25 @@ func stringSetFromPointers(in []*string) sets.String {
 	}
 	return out
 }
+
+// GetSourceAcctAndArn constructs source acct and arn and return them for use
+func GetSourceAcctAndArn(roleARN, region, clusterName string) (string, string, error) {
+	// ARN format (https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html)
+	// arn:partition:service:region:account-id:resource-type/resource-id
+	// IAM format, region is always blank
+	// arn:aws:iam::account:role/role-name-with-path
+	if !arn.IsARN(roleARN) {
+		return "", "", fmt.Errorf("incorrect ARN format for role %s", roleARN)
+	}
+	if region == "" {
+		return "", "", errors.New("region can't be empty")
+	}
+
+	parsedArn, err := arn.Parse(roleARN)
+	if err != nil {
+		return "", "", err
+	}
+
+	sourceArn := fmt.Sprintf("arn:%s:eks:%s:%s:cluster/%s", parsedArn.Partition, region, parsedArn.AccountID, clusterName)
```

Comment:
i understand that, in case of other use cases there would not be a valid aws resource right which would mean the source arn is invalid anyway?

Reply:
Correct, there's no single ARN that represents a non-EKS Kubernetes cluster. Should the request header not be set in that case?

Reply:
the way i see for this to be generic is make the arn generic `"arn:aws:*:us-west-2:1112222:cluster/test` such that we don't mention of eks in it. Since we don't differentiate whether its eks or not currently in the code, i don't think we should add that clause.

Reply:
That's fair, and I don't actually know how this request header is used so I don't know the impact of making the ARN generic. 👍🏻

Reply:
I think it’d be better to omit the header if this isn’t running in an EKS cluster, rather than using a kind of faked ARN. Wouldn’t an ECS cluster match that pattern too? The ambiguity is odd.

Reply:
yes, i was reading a bit more and i think it should be okay to pass the source arn as a param to the cloud provider along with the global arn since the aws docs also mention of using these values https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html

Reply:
@cartermckinnon @rifelpet can you check the new changes ?